src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -255,6 +297,27 @@ void ProtocolHandlerImpl::SendStartSessionAck(
       strncpy(protocolVersionString, (*minVersion).to_string().c_str(), 255);
       bson_object_put_string(
           &payloadObj, "protocolVersion", protocolVersionString);
+    } else if (serviceTypeValue == kMobileNav && additional_params != NULL) {
```

Comment:
I totally agree. I was hesitated to make changes on other feature's implementation. Anyway I implemented it in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/f6ea1262c4b6a3d1e10dc89fa570720a6c85ea9d, could you review it and see if it aligns with your thought? Thanks!